File: Easy/two-number-sum/solution_1/two-number-sum.js
// SOLUTION 1
// Two Number Sum
// Complexity: O(n^2) time | O(1) space

function twoNumberSum(array, targetSum) {
  const arrLen = array.length;
  for (let i = 0; i < arrLen - 1; i++) {
    const firstNum = array[i];
    for (let j = i + 1; j < arrLen; j++) {
      const secondNum = array[j];
      if (firstNum + secondNum === targetSum) {
        return [firstNum, secondNum];
      }
    }
  }
  return [];
}

File: Easy/two-number-sum/solution_1/two-number-sum.py
# SOLUTION 1
# Two Number Sum

# Complexity
# Average:  Time: O(n^2) | Space: O(1)
# Worst:    Time: | Space:

def twoNumberSum(array, targetSum):
    for i in range(len(array) - 1):
      firstNum = array[i]
      for j in range(i + 1, len(array)):
        secondNum = array[j]
        if firstNum + secondNum == targetSum:
          return [firstNum, secondNum]
    return []